pkg/cloud/gcp/assets/validating-admission-policy.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: network-tier-annotation-validation-policy
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups: [""]
+        apiVersions: ["v1"]
+        operations: ["CREATE", "UPDATE"]
+        resources: ["services"]
+  variables:
+    - name: hasNetworkTier
+      expression: has(object.metadata.annotations) && 'cloud.google.com/network-tier' in object.metadata.annotations
+    - name: networkTierValue
+      expression: object.metadata.?annotations['cloud.google.com/network-tier'].orValue("")
+    - name: hasOldNetworkTier
+      expression: has(oldObject) && has(oldObject.metadata) && has(oldObject.metadata.annotations) && 'cloud.google.com/network-tier' in oldObject.metadata.annotations
+    - name: oldNetworkTierValue
+      expression: oldObject.metadata.?annotations['cloud.google.com/network-tier'].orValue("")
+    - name: valueHasChanged
+      expression: |
```

Comment:
Hmm, perhaps actually, valueHasChanged should be

```
variables.hasNetworkTier && (!variables.hasOldNetworkTier || variables.oldNetworkTierValue != variables.networkTierValue)
```

Then below for the validations, it can simply be
```
!variables.valueHasChanged || variables.networkTierValue in ['Standard', 'Premium']
```

As the valueHasChanged will show true only when we definitely have a network tier

Reply:
This will result in early exit , shall we prefer this ? 
`!variables.hasOldNetworkTier ||
        variables.hasNetworkTier &&
        variables.oldNetworkTierValue != variables.networkTierValue`

Reply:
Validation I think we can update to what you are suggesting .

Reply:
But I think the early exit is fine. If we don't have the network tier value now, there's no point continuing, so we return false, and the `!variables.hasValueChanged` returns true, exiting the later validation, which is appropriate